types/src/system/auction/bid/vesting.rs

```diff
@@ -151,104 +153,87 @@ impl VestingSchedule {
     }
 }
 
-impl ToBytes for VestingSchedule {
-    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+impl ToBytes for [U512; LOCKED_AMOUNTS_MAX_LENGTH] {
+    fn to_bytes(&self) -> Result<Vec<u8>, Error> {
         let mut result = bytesrepr::allocate_buffer(self)?;
-        self.write_bytes(&mut result)?;
+        for item in self.iter() {
+            result.append(&mut item.to_bytes()?);
+        }
         Ok(result)
     }
 
     fn serialized_length(&self) -> usize {
-        self.initial_release_timestamp_millis.serialized_length()
-            + LOCKED_AMOUNTS_TAG_LENGTH
-            + match &self.locked_amounts {
-                Some(locked_amounts) => locked_amounts.serialized_length(),
-                None => 0,
-            }
+        self.iter().map(ToBytes::serialized_length).sum::<usize>()
     }
+}
 
-    fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), bytesrepr::Error> {
-        self.initial_release_timestamp_millis.write_bytes(writer)?;
-        match &self.locked_amounts {
-            None => {
-                UNINITIALIZED_LOCKED_AMOUNTS_TAG.write_bytes(writer)?;
-            }
-            Some(locked_amounts) => {
-                VARIABLE_LOCKED_AMOUNTS_LENGTH_TAG.write_bytes(writer)?;
-                locked_amounts.write_bytes(writer)?;
-            }
+impl FromBytes for [U512; LOCKED_AMOUNTS_MAX_LENGTH] {
+    fn from_bytes(mut bytes: &[u8]) -> Result<(Self, &[u8]), Error> {
+        let mut result: MaybeUninit<[U512; LOCKED_AMOUNTS_MAX_LENGTH]> = MaybeUninit::uninit();
```

Comment:
Let's avoid `MaybeUninit`.

Reply:
Done, and moved the serialization logic to `fn write_bytes`